### PR TITLE
Allow multiple insertions of #reason

### PR DESCRIPTION
**Pull request by mgramigna**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

For ASCODCP-1936. Originally scoped as writing a completion component for multi-selection, but shifted towards allowing #reason to be inserted multiple times using the same completion component as single-choice shortcuts.

Can now enter #reason to select a value. Can enter as many #reason shortcuts and select as you want until you have selected all choices in the value set.

**NOTE**: While doing this task I uncovered something on `master`/fnorg where the value set options for #reasons are always in the suggestion portal regardless of if they are in the editor or not. Created ASCODCP-2104 as a result.

Potential thing to keep in mind for the future: With this approach, I think clinician customization of the expanded text will need to support adding multiple incomplete #reason labels since we need to support 0 .. many. 

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by jafeltra** [comment]
This looks good! I like this approach and I personally don't find adding multiple #reason's too confusing. Two things I noticed that may or may not be in scope of this PR:

[JA1] Add "#reason (uncompleted) some other text #pathology (added in however you want)" now if you go back to #reason to complete it, Pathology is still in the portal of suggestions, even though it is already in the editor. Complete to pathology and it can be set twice, but only shows up once in the TDP.
[JA2] After completing the steps in JA1, if you go to the end of the line and delete the second "Pathology," the TDP shows no data for the rationale, even though there is a pathology added to the note.

**Comment by sgober** [comment]
[SO1] I am getting an error when I type out what is in the screenshot below, then move the cursor up next to the condition and then back down to the end of the last reason

Screen Shot 2019-07-01 at 9 07 08 AM

Screen Shot 2019-07-01 at 9 07 15 AM

**Comment by mgramigna** [comment]
@sgober SO1 is already a Jira that we found in #665

**Comment by sgober** [comment]
@mgramigna my b, too many JIRAs for me to keep track of

**Comment by mgramigna** [comment]
That Jira that I mentioned that I found in the PR could also encompass JA2: https://jira.mitre.org/browse/ASCODCP-2104 since that is the root reason that we can type multiple pathologies in the first place.

Looking into JA1 now

**Comment by mgramigna** [comment]
*TL;DR: JA1 = Jira?*

---

So for JA1. This happens because the value selection options are only set on initialize. When you insert #reason, the valueSet that it uses is determined on initialize. When you then insert #pathology, it never updates the valueSet from #reason since it doesn't need to get re-initialized.

Simplest way would be to have the valueSet re-compute whenever the completion portal opens? There could be a more at-large refactor here for the shortcut model to make this work nicely. Either way, I'm gonna go on a limb and say it shouldn't be resolved for this PR. Thoughts?

**Comment by mgramigna** [comment]
Also, JA1 might not even be a use case in the MVP in my opinion

**Comment by jafeltra** [comment]
I agree. The root cause of it seems separate from this task so I think we're okay to not include it in this PR and make a Jira.
